web_src/js/components/RepoBranchTagSelector.vue

```diff
@@ -246,7 +246,35 @@ const sfc = {
         event.preventDefault();
         this.menuVisible = false;
       }
-    }
+    },
+    handleTabSwitch(mode) {
+      this.createTag = mode === 'tags';
+      this.mode = mode;
+      this.focusSearchField();
+      this.fetchBranchesOrTags();
+    },
+    async fetchBranchesOrTags() {
+      if (!['branches', 'tags'].includes(this.mode) || this.isLoading) return;
+      // only fetch when branch/tag list has not been initialized
+      if (this.hasListInitialized[this.mode] ||
+      (this.mode === 'branches' && !this.showBranchesInDropdown) || (this.mode === 'tags' && this.noTag)) return;
```

Comment:
Need to reformat

Reply:
Is there specific suggestion for reformat?

Reply:
```suggestion
      if (this.hasListInitialized[this.mode] ||
        (this.mode === 'branches' && !this.showBranchesInDropdown) ||
        (this.mode === 'tags' && this.noTag)
      ) {
        return;
      }
```